internal/controller/multiclusterservice_controller.go

```diff
@@ -337,7 +337,31 @@ func (r *MultiClusterServiceReconciler) setClustersServicesReadinessConditions(c
 		}
 	}
 
-	desiredClusters, desiredServices := len(clusters.Items), len(clusters.Items)*len(mcs.Spec.ServiceSpec.Services)
+	// There is also SveltosCluster type to consider.
+	sveltosClusters := &metav1.PartialObjectMetadataList{}
+	sveltosClusters.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   "lib.projectsveltos.io",
+		Version: "v1beta1",
+		Kind:    "SveltosCluster",
+	})
+	if err := r.Client.List(ctx, sveltosClusters, client.MatchingLabelsSelector{Selector: sel}); err != nil {
+		return fmt.Errorf("failed to list partial SveltosClusters: %w", err)
+	}
+
+	for _, cluster := range sveltosClusters.Items {
+		key := client.ObjectKey{Namespace: cluster.Namespace, Name: cluster.Name}
+		sc := libsveltosv1beta1.SveltosCluster{}
+		if err := r.Client.Get(ctx, key, &sc); err != nil {
+			return fmt.Errorf("failed to get SveltosCluster %s: %w", key.String(), err)
+		}
```

Comment:
i believe we could just fetch the full resource list, without making extra API calls, considering we already have a selector that reduces the number of objects to be listed